contracts/governance/FeeSharingCollector/FeeSharingCollector.sol

```diff
@@ -344,42 +347,52 @@ contract FeeSharingCollector is
-        uint256 fromCheckpointIndex = _fromCheckpoint - 1;
+    function validFromCheckpointParam(
+        uint256[] memory _fromCheckpoints,
+        address[] memory _tokens,
+        address _user
+    ) private view {
         require(
-            _fromCheckpoint > processedCheckpoints[_user][_token],
-            "_fromCheckpoint param must be > userProcessedCheckpoints"
-        );
-        require(
-            _fromCheckpoint <= totalTokenCheckpoints[_token],
-            "_fromCheckpoint should be <= totalTokenCheckpoints"
+            _tokens.length == _fromCheckpoints.length,
+            "mismatch tokens and checkpoints length"
         );
 
-        Checkpoint memory prevCheckpoint = tokenCheckpoints[_token][fromCheckpointIndex - 1];
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            // _fromCheckpoint is checkpoint number, not array index, so should be > 1
+            require(_fromCheckpoints[i] > 1, "_fromCheckpoint param must be > 1");
+            uint256 fromCheckpointIndex = _fromCheckpoints[i] - 1;
+            require(
+                _fromCheckpoints[i] > processedCheckpoints[_user][_tokens[i]],
+                "_fromCheckpoint param must be > userProcessedCheckpoints"
+            );
+            require(
+                _fromCheckpoints[i] <= totalTokenCheckpoints[_tokens[i]],
+                "_fromCheckpoint should be <= totalTokenCheckpoints"
+            );
 
-        uint96 weightedStake =
-            staking.getPriorWeightedStake(
-                _user,
-                prevCheckpoint.blockNumber - 1,
-                prevCheckpoint.timestamp
+            Checkpoint memory prevCheckpoint =
+                tokenCheckpoints[_tokens[i]][fromCheckpointIndex - 1];
+
+            uint96 weightedStake =
+                staking.getPriorWeightedStake(
+                    _user,
+                    prevCheckpoint.blockNumber - 1,
+                    prevCheckpoint.timestamp
+                );
+            require(
+                weightedStake == 0,
+                "User weighted stake should be zero at previous checkpoint"
             );
-        require(weightedStake == 0, "User weighted stake should be zero at previous checkpoint");
 
-        Checkpoint memory fromCheckpoint = tokenCheckpoints[_token][fromCheckpointIndex];
-        weightedStake = staking.getPriorWeightedStake(
-            _user,
-            fromCheckpoint.blockNumber - 1,
-            fromCheckpoint.timestamp
-        );
+            Checkpoint memory fromCheckpoint = tokenCheckpoints[_tokens[i]][fromCheckpointIndex];
+            weightedStake = staking.getPriorWeightedStake(
+                _user,
+                fromCheckpoint.blockNumber - 1,
+                fromCheckpoint.timestamp
+            );
 
-        require(weightedStake > 0, "User weighted stake should be > 0 at  _fromCheckpoint");
-        _;
+            require(weightedStake > 0, "User weighted stake should be > 0 at  _fromCheckpoint");
+        }
```

Comment:
I think we can save some gas and drop this.

Reply:
i'd prefer not to remove this check since we can save more gas in case we got wrong checkpoint with 0 amount

Reply:
only if there is a mistake in calculating "from" checkpoints which are pre-calculated, so "empty" checkpoints with 0 weight should not happen or be a very rare case.  
removing this should save gas on all withdrawals which is a preference for users to save on gas because they'd love to have more checkpoints processed at a time.